File: android/src/main/kotlin/com/example/downloader_plugin/models/ResumeData.kt
package com.example.downloader_plugin.models

/// Holds data associated with a resume
class ResumeData(val task: Task, val data: String, val requiredStartByte: Long) {
    fun toJsonMap(): MutableMap<String, Any?> {
        return mutableMapOf(
            "task" to task.toJsonMap(),
            "data" to data,
            "requiredStartByte" to requiredStartByte
        )
    }
}

File: lib/models/resume_data.dart
import 'task.dart';

/// Holds data associated with a resume
class ResumeData {
  final Task task;
  final String data;
  final int requiredStartByte;

  ResumeData(this.task, this.data, this.requiredStartByte);

  /// Create object from JSON Map
  ResumeData.fromJsonMap(Map<String, dynamic> jsonMap)
      : task = Task.createFromJsonMap(jsonMap['task']),
        data = jsonMap['data'] as String,
        requiredStartByte = jsonMap['requiredStartByte'] as int;

  /// Return JSON Map representing object
  Map<String, dynamic> toJsonMap() => {'task': task.toJsonMap(), 'data': data, 'requiredStartByte': requiredStartByte};

  String get taskId => task.taskId;
}
